peer/peer.go

```diff
@@ -1348,6 +1349,13 @@ out:
 				continue
 			}
 
+			// Ignore unknown messages to ease future extensibility.
```

Comment:
Ideally when _creating_ the peer object itself, the caller is able to specify if they wish to permit unknown message or not, as the caller may wish to adhere to a stricter interpretation of what exactly constitutes the Bitcoin P2P protocol. 

Also it seems this change should wait on the development of this draft BIP: https://github.com/sdaftuar/bips/blob/2020-08-generalized-feature-negotiation/bip-p2p-feature-negotiation.mediawiki

Reply:
I think the comment is a reference to https://github.com/benjyz/bitcoinArchive/blob/master/bitcoin0.1/src/main.cpp#L2035-L2039. But I agree for the library this should be extensible